Hearthstone Deck Tracker/Hearthstone/GameV2.cs

```diff
@@ -74,7 +74,7 @@ public bool IsOpponentMinionInPlay
 			=> Entities.FirstOrDefault(x => (x.Value.IsInPlay && x.Value.IsMinion && x.Value.IsControlledBy(Opponent.Id))).Value != null;
 
 		public int OpponentMinionCount => Entities.Count(x => (x.Value.IsInPlay && x.Value.IsMinion && x.Value.IsControlledBy(Opponent.Id)));
-		public int PlayerMinionCount => Entities.Count(x => (x.Value.IsInPlay && x.Value.IsMinion && x.Value.IsControlledBy(Player.Id)));
+		public int PlayerMinionCount => Entities.Count(x => (x.Value.IsInPlay && x.Value.IsMinion && x.Value.IsControlledBy(Player.Id) && !x.Value.HasTag(GameTag.TO_BE_DESTROYED)));
```

Comment:
I'm not sure if changing this is the best thing to do. It might break other stuff depending on it.

Reply:
Yes, this might need some discussion. As far as I know, minions marked as "TO_BE_DESTROYED" will be moved to GRAVEYARD very soon. So it seems OK to see them as if they died.

Here's the log when 'Brawl' is played and minions get killed.

`D 20:50:23.2108479 PowerTaskList.DebugPrintPower() - BLOCK_START BlockType=PLAY Entity=[entityName=Brawl id=63 zone=HAND zonePos=4 cardId=EX1_407 player=2] EffectCardId= EffectIndex=0 Target=0 SubOption=-1 
D 20:50:23.2108479 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=LuckyPants#5910 tag=RESOURCES_USED value=5 
D 20:50:23.2108479 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=LuckyPants#5910 tag=NUM_RESOURCES_SPENT_THIS_GAME value=28 
D 20:50:23.2108479 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=LuckyPants#5910 tag=NUM_CARDS_PLAYED_THIS_TURN value=1 
D 20:50:23.2108479 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=LuckyPants#5910 tag=430 value=1 
D 20:50:23.2108479 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Brawl id=63 zone=HAND zonePos=4 cardId=EX1_407 player=2] tag=1068 value=1 
D 20:50:23.2108479 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Brawl id=63 zone=HAND zonePos=4 cardId=EX1_407 player=2] tag=1068 value=0 
D 20:50:23.2118480 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Shield Block id=43 zone=HAND zonePos=9 cardId=EX1_606 player=2] tag=ZONE_POSITION value=8 
D 20:50:23.2118480 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Witchwood Piper id=62 zone=HAND zonePos=8 cardId=GIL_584 player=2] tag=ZONE_POSITION value=7 
D 20:50:23.2118480 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Commanding Shout id=44 zone=HAND zonePos=7 cardId=NEW1_036 player=2] tag=ZONE_POSITION value=6 
D 20:50:23.2118480 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Restless Mummy id=40 zone=HAND zonePos=6 cardId=ULD_206 player=2] tag=ZONE_POSITION value=5 
D 20:50:23.2118480 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Rabid Worgen id=38 zone=HAND zonePos=5 cardId=GIL_113 player=2] tag=ZONE_POSITION value=4 
D 20:50:23.2118480 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Brawl id=63 zone=HAND zonePos=4 cardId=EX1_407 player=2] tag=1037 value=0 
D 20:50:23.2118480 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Brawl id=63 zone=HAND zonePos=4 cardId=EX1_407 player=2] tag=ZONE_POSITION value=0 
D 20:50:23.2118480 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Brawl id=63 zone=HAND zonePos=4 cardId=EX1_407 player=2] tag=ZONE value=PLAY 
D 20:50:23.2118480 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Brawl id=63 zone=HAND zonePos=4 cardId=EX1_407 player=2] tag=JUST_PLAYED value=1 
D 20:50:23.2118480 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=LuckyPants#5910 tag=LAST_CARD_PLAYED value=63 
D 20:50:23.2118480 PowerTaskList.DebugDump() - Block End=(null)
D 20:50:23.2118480 PowerProcessor.PrepareHistoryForCurrentTaskList() - m_currentTaskList=394
D 20:50:23.2118480 PowerProcessor.DoTaskListForCard() - unhandled BlockType PLAY for sourceEntity [entityName=Brawl id=63 zone=HAND zonePos=4 cardId=EX1_407 player=2]
D 20:50:23.2188484 PowerProcessor.EndCurrentTaskList() - m_currentTaskList=394
D 20:50:23.2338492 PowerTaskList.DebugDump() - ID=395 ParentID=394 PreviousID=0 TaskCount=13
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() - BLOCK_START BlockType=POWER Entity=[entityName=Brawl id=63 zone=PLAY zonePos=0 cardId=EX1_407 player=2] EffectCardId= EffectIndex=0 Target=0 SubOption=-1 
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     META_DATA - Meta=TARGET Data=0 InfoCount=1
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -                 Info[0] = [entityName=Wild Bloodstinger id=5 zone=PLAY zonePos=1 cardId=ULD_212 player=1]
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     META_DATA - Meta=TARGET Data=0 InfoCount=1
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -                 Info[0] = [entityName=Loot Hoarder id=60 zone=PLAY zonePos=1 cardId=EX1_096 player=2]
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     META_DATA - Meta=TARGET Data=0 InfoCount=1
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -                 Info[0] = [entityName=Springpaw id=18 zone=PLAY zonePos=2 cardId=TRL_348 player=1]
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     META_DATA - Meta=TARGET Data=0 InfoCount=1
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -                 Info[0] = [entityName=Lynx id=98 zone=PLAY zonePos=3 cardId=TRL_348t player=1]
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     META_DATA - Meta=TARGET Data=0 InfoCount=1
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -                 Info[0] = [entityName=Wyvern id=102 zone=PLAY zonePos=4 cardId=DAL_378t1 player=1]
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wild Bloodstinger id=5 zone=PLAY zonePos=1 cardId=ULD_212 player=1] tag=TO_BE_DESTROYED value=1 
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wild Bloodstinger id=5 zone=PLAY zonePos=1 cardId=ULD_212 player=1] tag=LAST_AFFECTED_BY value=63 
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Loot Hoarder id=60 zone=PLAY zonePos=1 cardId=EX1_096 player=2] tag=TO_BE_DESTROYED value=1 
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Loot Hoarder id=60 zone=PLAY zonePos=1 cardId=EX1_096 player=2] tag=LAST_AFFECTED_BY value=63 
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Lynx id=98 zone=PLAY zonePos=3 cardId=TRL_348t player=1] tag=TO_BE_DESTROYED value=1 
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Lynx id=98 zone=PLAY zonePos=3 cardId=TRL_348t player=1] tag=LAST_AFFECTED_BY value=63 
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wyvern id=102 zone=PLAY zonePos=4 cardId=DAL_378t1 player=1] tag=TO_BE_DESTROYED value=1 
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wyvern id=102 zone=PLAY zonePos=4 cardId=DAL_378t1 player=1] tag=LAST_AFFECTED_BY value=63 
D 20:50:23.2338492 PowerTaskList.DebugPrintPower() - BLOCK_END
D 20:50:23.2338492 PowerProcessor.PrepareHistoryForCurrentTaskList() - m_currentTaskList=395
D 20:50:30.2582510 PowerProcessor.EndCurrentTaskList() - m_currentTaskList=395
D 20:50:30.2732519 PowerTaskList.DebugDump() - ID=396 ParentID=0 PreviousID=394 TaskCount=3
D 20:50:30.2732519 PowerTaskList.DebugDump() - Block Start=(null)
D 20:50:30.2732519 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Brawl id=63 zone=PLAY zonePos=0 cardId=EX1_407 player=2] tag=1068 value=4 
D 20:50:30.2732519 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Brawl id=63 zone=PLAY zonePos=0 cardId=EX1_407 player=2] tag=1068 value=0 
D 20:50:30.2732519 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Brawl id=63 zone=PLAY zonePos=0 cardId=EX1_407 player=2] tag=ZONE value=GRAVEYARD 
D 20:50:30.2732519 PowerTaskList.DebugDump() - Block End=(null)
D 20:50:30.2732519 PowerProcessor.PrepareHistoryForCurrentTaskList() - m_currentTaskList=396
D 20:50:30.2732519 PowerProcessor.DoTaskListForCard() - unhandled BlockType PLAY for sourceEntity [entityName=Brawl id=63 zone=PLAY zonePos=0 cardId=EX1_407 player=2]
D 20:50:30.3242548 PowerProcessor.EndCurrentTaskList() - m_currentTaskList=396
D 20:50:30.3402557 PowerTaskList.DebugDump() - ID=397 ParentID=394 PreviousID=0 TaskCount=44
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() - BLOCK_START BlockType=DEATHS Entity=GameEntity EffectCardId= EffectIndex=0 Target=0 SubOption=-1 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wild Bloodstinger id=5 zone=PLAY zonePos=1 cardId=ULD_212 player=1] tag=TO_BE_DESTROYED value=0 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=FightPants#5170990 tag=NUM_MINIONS_PLAYER_KILLED_THIS_TURN value=1 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=GameEntity tag=NUM_MINIONS_KILLED_THIS_TURN value=1 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wild Bloodstinger id=5 zone=PLAY zonePos=1 cardId=ULD_212 player=1] tag=1068 value=4 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wild Bloodstinger id=5 zone=PLAY zonePos=1 cardId=ULD_212 player=1] tag=1068 value=0 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wild Bloodstinger id=5 zone=PLAY zonePos=1 cardId=ULD_212 player=1] tag=EXHAUSTED value=0 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wyvern id=102 zone=PLAY zonePos=4 cardId=DAL_378t1 player=1] tag=ZONE_POSITION value=3 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Lynx id=98 zone=PLAY zonePos=3 cardId=TRL_348t player=1] tag=ZONE_POSITION value=2 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Springpaw id=18 zone=PLAY zonePos=2 cardId=TRL_348 player=1] tag=ZONE_POSITION value=1 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wild Bloodstinger id=5 zone=PLAY zonePos=1 cardId=ULD_212 player=1] tag=ZONE_POSITION value=0 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wild Bloodstinger id=5 zone=PLAY zonePos=1 cardId=ULD_212 player=1] tag=ZONE value=GRAVEYARD 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=FightPants#5170990 tag=NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_TURN value=1 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=FightPants#5170990 tag=NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_GAME value=3 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wild Bloodstinger id=5 zone=PLAY zonePos=1 cardId=ULD_212 player=1] tag=DAMAGE value=0 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Loot Hoarder id=60 zone=PLAY zonePos=1 cardId=EX1_096 player=2] tag=TO_BE_DESTROYED value=0 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=LuckyPants#5910 tag=NUM_MINIONS_PLAYER_KILLED_THIS_TURN value=1 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=GameEntity tag=NUM_MINIONS_KILLED_THIS_TURN value=2 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Loot Hoarder id=60 zone=PLAY zonePos=1 cardId=EX1_096 player=2] tag=1068 value=4 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Loot Hoarder id=60 zone=PLAY zonePos=1 cardId=EX1_096 player=2] tag=1068 value=0 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Loot Hoarder id=60 zone=PLAY zonePos=1 cardId=EX1_096 player=2] tag=ZONE_POSITION value=0 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Loot Hoarder id=60 zone=PLAY zonePos=1 cardId=EX1_096 player=2] tag=ZONE value=GRAVEYARD 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=LuckyPants#5910 tag=NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_TURN value=1 
D 20:50:30.3402557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=LuckyPants#5910 tag=NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_GAME value=7 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Lynx id=98 zone=PLAY zonePos=3 cardId=TRL_348t player=1] tag=TO_BE_DESTROYED value=0 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=FightPants#5170990 tag=NUM_MINIONS_PLAYER_KILLED_THIS_TURN value=2 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=GameEntity tag=NUM_MINIONS_KILLED_THIS_TURN value=3 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Lynx id=98 zone=PLAY zonePos=3 cardId=TRL_348t player=1] tag=1068 value=4 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Lynx id=98 zone=PLAY zonePos=3 cardId=TRL_348t player=1] tag=1068 value=0 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Lynx id=98 zone=PLAY zonePos=3 cardId=TRL_348t player=1] tag=EXHAUSTED value=0 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wyvern id=102 zone=PLAY zonePos=4 cardId=DAL_378t1 player=1] tag=ZONE_POSITION value=2 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Lynx id=98 zone=PLAY zonePos=3 cardId=TRL_348t player=1] tag=ZONE_POSITION value=0 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Lynx id=98 zone=PLAY zonePos=3 cardId=TRL_348t player=1] tag=ZONE value=GRAVEYARD 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=FightPants#5170990 tag=NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_TURN value=2 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=FightPants#5170990 tag=NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_GAME value=4 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wyvern id=102 zone=PLAY zonePos=4 cardId=DAL_378t1 player=1] tag=TO_BE_DESTROYED value=0 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=FightPants#5170990 tag=NUM_MINIONS_PLAYER_KILLED_THIS_TURN value=3 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=GameEntity tag=NUM_MINIONS_KILLED_THIS_TURN value=4 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wyvern id=102 zone=PLAY zonePos=4 cardId=DAL_378t1 player=1] tag=1068 value=4 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wyvern id=102 zone=PLAY zonePos=4 cardId=DAL_378t1 player=1] tag=1068 value=0 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wyvern id=102 zone=PLAY zonePos=4 cardId=DAL_378t1 player=1] tag=1037 value=0 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wyvern id=102 zone=PLAY zonePos=4 cardId=DAL_378t1 player=1] tag=ZONE_POSITION value=0 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=[entityName=Wyvern id=102 zone=PLAY zonePos=4 cardId=DAL_378t1 player=1] tag=ZONE value=GRAVEYARD 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=FightPants#5170990 tag=NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_TURN value=3 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() -     TAG_CHANGE Entity=FightPants#5170990 tag=NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_GAME value=5 
D 20:50:30.3412557 PowerTaskList.DebugPrintPower() - BLOCK_END`